### ISSUE TITLE
Custom markers don't always disappear even though they are being counted in the cluster

### ISSUE THREAD

**Issue by pdugan20** [venits/react-native-map-clustering#171]
![animation5](https://user-images.githubusercontent.com/1837100/83098795-6761f900-a060-11ea-9300-a8b27da763b4.gif)

Implementation is as follows, happy to share more if needed.

```
<MapView
    ref={this.mapRef}
    style={{ height: '100%' }}
    initialRegion={region}
    showsUserLocation
    showsMyLocationButton={false}
    showsPointsOfInterest={false}
    showsCompass
    clusterColor={colors.purple}
    radius={radius}
    animationEnabled={false}
    onRegionChange={this.onRegionChange}
    onRegionChangeComplete={this.onRegionChangeComplete}
    loadingIndicatorColor={theme.colors.loadingSpinner}
    loadingBackgroundColor={theme.colors.mapViewBackground}
>
    {hikeData.map(({ id, coordinates, distance }, index) => (
        <HikeMapMarker
            key={id}
            identifier={id}
            distance={distance}
            markerRef={(ref) => this.assignRef(ref, index)}
            coordinate={{
                latitude: coordinates.center.lat,
                longitude: coordinates.center.lng,
            }}
            onPress={this.markerPress}
            tracksViewChanges={tracksViewChanges}
        />
    ))}
</MapView>
```

**Comment by pdugan20** [comment]
Upgraded to 3.3.9 and appears to now be worse? Markers are now also disappearing at random and the clusters don't appear on region change until I actually move the map around after zooming in or out.

Happy to provide more code samples if that's helpful.

![animation5](https://user-images.githubusercontent.com/1837100/83184443-5526a000-a0de-11ea-869c-0e66e71cf79a.gif)

**Comment by pdugan20** [comment]
Anything I can do here to provide additional context or code samples?

**Comment by venits** [comment]
Fixed in version `3.4.0`
